poet_src/Evolve/BinarySystem.cpp

```diff
@@ -411,7 +411,7 @@ namespace Evolve {
                                                 double orbit_angmom_gain_deriv,
                                                 bool semimajor_deriv) const
     {
-        if(__eccentricity == 0) return 0;
+        if(__eccentricity <= 1e-8) return 0;
```

Comment:
I don't like the hard-coded parameter that we have no control over. These tend to be forgotten and then we will scratch our heads as to why something is happening. I see 3 options:

  1. Specify this limit when the system is created, i.e. as an argument to the constructor, and probably set it to the precision that we use.

  2. Tune it automatically based on the sum of the magnitudes of the two terms that cancel at zero eccentricity. Something like when |(P/E) + (2*T/L)| < 1e-14 * (|P/E| + |2 * T/L|), where P is the tidal power, T is the torque, E is the orbital energy and L is the orbital angular momentum. This would mean that he numerical round-off error is within 2 orders of magnitude of the value of the rate of circularization.

3. Same as either 1 or 2 but when this happens also set the system eccentricty to exactly zero. This is probably the best option, but it also involves the most work.